quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusCodeActions.java

```diff
@@ -184,37 +197,27 @@ private void doCodeActionsForUnknownEnumValue(Diagnostic diagnostic, PropertiesM
 	}
 
 	/**
-	 * Returns the <code>Position</code> to insert the missing required code action property into
-	 * @param textDocument the text document
-	 * @return the <code>Position</code> to insert the missing required code action property into
-	 * @throws BadLocationException
+	 * Create a code action that adds <code>propertyName</code> to user's unknown
+	 * validation excluded array
+	 * 
+	 * @param propertyName the property name to add to array for code action
+	 * @param diagnostic   the corresponding unknown property diagnostic
+	 * @param codeActions  the list of code actions
 	 */
-	private Position getPositionForRequiredCodeAction(TextDocument textDocument) throws BadLocationException {
-		String textDocumentText = textDocument.getText();
-		
-		if (!StringUtils.hasText(textDocumentText)) {
-			return new Position(0, 0);
-		}
+	private void doCodeActionForIgnoreUnknownValidation(String propertyName, Diagnostic diagnostic,
+			List<CodeAction> codeActions) {
+		CodeAction insertContentAction = new CodeAction(
+				"Exclude '" + propertyName + "' from unknown property validation?");
 
-		for (int i = textDocumentText.length() - 1; i >= 0; i--) {
-			if (!Character.isWhitespace(textDocumentText.charAt(i))) {
-				return textDocument.positionAt(i + 1);
-			}
-		}
-
-		// should never happen
-		return null;
-	}
+		ConfigurationItemEdit configItemEdit = new ConfigurationItemEdit("quarkus.tools.validation.unknown.excluded",
+				ConfigurationItemEditType.add, propertyName);
 
-	/**
-	 * Returns the missing required property name from <code>diagnosticMessage</code>
-	 * @param diagnosticMessage the diagnostic message containing the property name in single quotes
-	 * @return the missing required property name from <code>diagnosticMessage</code>
-	 */
-	private String getPropertyNameFromRequiredMessage(String diagnosticMessage) {
-		int start = diagnosticMessage.indexOf('\'') + 1;
-		int end = diagnosticMessage.indexOf('\'', start);
-		return diagnosticMessage.substring(start, end);
+		Command command = new Command("Add " + propertyName + " to unknown excluded array",
+				"quarkus/configurationUpdate", Arrays.asList(configItemEdit));

```

Comment:
"quarkus/configurationUpdate" should be replaced with CommandKind.COMMAND_CONFIGURATION_UPDATE, no?

Reply:
replace `Arrays.asList(configItemEdit)` with `Collections.singletonList(configItemEdit)`